src/t_stream.c

```diff
@@ -1532,6 +1721,17 @@ size_t streamReplyWithRange(client *c, stream *s, streamID *start, streamID *end
     while(streamIteratorGetID(&si,&id,&numfields)) {
         /* Update the group last_id if needed. */
         if (group && streamCompareID(&id,&group->last_id) > 0) {
+            if (group->entries_read && streamRangeDoesNotContainTombstone(s,&id)) {
```

Comment:
so after our discussion, an "invalid" group->entries_read is UINT64_MAX rather than 0, right?

Reply:
Yep.